crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -34,7 +34,8 @@ class ApplicationController < ActionController::Base
     Rails.env.test?
   }
   before_filter :upgrade, if: proc {
-    File.exist?("/var/lib/crowbar/upgrade/7-to-8-upgrade-running")
+    File.exist?("/var/lib/crowbar/upgrade/7-to-8-upgrade-running") &&
+      !File.exist?("/var/lib/crowbar/upgrade/7-to-8-upgrade-compute-nodes-postponed")
```

Comment:
Does this imply that the crowbar's full functionality is available when the compute nodes upgrade is postponed? Sounds a little dangerous. E.g what happens if the nova barclamp is applied during this timeframe?

Reply:
Yes, it is dangerous when the barclamp apply affects compute nodes. But that's how I understand the requirement.

There's actually no other effect of this postponing than allowing users to access the barclamps view. If we'd want to restrict it until the full upgrade is done, there's no need for postpone action. We could just enhance UI a little bit so it allows what current CLI already has (separating controllers and computes upgrade).

Reply:
Hm, there's doesn't seem to be  anything in the spec that requires the barclamps to be changeable while there are some notes not upgraded yet. But we might need some clarification on that. As far as I understood it just was just about giving the user some more control about when to upgrade which (compute) nodes. (I think we implemented most of that is already for the C6 to 7 upgrade)

If we really want the barclamp to be editable after the controllers are upgrade I guess we should restrict the change to barclamps that don't have any roles assigned to non-upgraded compute nodes (which might a little involved I guess).

Reply:
> Hm, there's doesn't seem to be anything in the spec that requires the barclamps to be
> changeable while there are some notes not upgraded yet.

Maybe not in spec, but I remember this was @dirkmueller explanation

>  think we implemented most of that is already for the C6 to 7 upgrade

Indeed. Such limited functionality we already have.

> restrict the change to barclamps that don't have any roles assigned to non-upgraded compute > nodes 

Sounds reasonable, but might be tricky

Reply:
But this approach is indeed up to discussion

Reply:
Anyway, if this is the only problem, I'd leave it to followup PR